### ISSUE TITLE
There are no logs for Cassandra startup errors

### ISSUE THREAD

**Issue by jsanda** [k8ssandra/k8ssandra#680]
## Bug Report

**Describe the bug**
It is difficult to debug Cassandra startup errors. The primary process in the `cassandra` container is the management-api. It manages the lifecycle of Cassandra. When a Cassandra pod is running but not ready, this generally indicates that Cassandra has failed to start for some reason. Often times there are no logs available which makes debugging really difficult. 

I will provide an example using an invalid heap configuration.

**To Reproduce**
1. `helm install broken-cluster k8ssandra/k8ssandra -f test-values.yaml`

test-values.yaml:
```yaml
cassandra:
  heap:
    size: 768
    newGenSize: 256
  datacenters:
  - name: dc1
    size: 3
```

2. Wait for the `cassandra` container to start. It should not reach the ready state.
3. Do `kubectl logs broken-cluster-dc1-default-sts-0 -c server-system-logger`. It will  report:

`can't open '/var/log/cassandra/system.log': No such file or directory`

4. Do `kubectl exec -it broken-cluster-dc1-default-sts-0 -c cassandra -- ls /var/log/cassandra`. The directory is empty.

**Expected behavior**
Logs with the failure should be available. If I open a shell into the Cassandra pod and start Cassandra in the foreground with -f, I get the following output:

```
cassandra@broken-cluster-dc1-default-sts-0:/opt/cassandra/bin$ ./cassandra -f
Error occurred during initialization of VM
Too small initial heap
```
This ouptut should be captured in some log file.

* Helm charts version info
1.0, 1.1

**Comment by emerkle826** [comment]
@jsanda I'm going to try to get this fixed.

**Comment by emerkle826** [comment]
When I tried the above steps, with a C* cluster size of `1`, I do get the following files in `/var/log/cassandra`:

    gc.log
    stderr.log
    stdout.log

Neither `gc.log` nor `stderr.log` have any contents, but `stdout.log` does have:

    Error occurred during initialization of VM
    Too small initial heap

**Comment by emerkle826** [comment]
After investigating this further, it seems that only one Cassandra pod will generate log files in `/var/log/cassandra` under these conditions. That pod will try to restart since it does not start cleanly, and the restarts will overwrite the log files (but with the same contents). It's not always pod 0 that gets the logs either.

**Comment by jsanda** [comment]
You are correct. I was hasty and only look at one of the pods. The reason the logs files are present on only of the pods is that cass-operator starts the Cassandra nodes serially and waits until the C* node is initialized before starting the next one. 

Having the log files overwritten is annoying but we can work with it since as you mentioned the contents will be the same. For now I think some documentation around this would be good.

**Comment by emerkle826** [comment]
Yes, docs around this would help. I was also considering redirecting stderr to stdout and just have 1 file to check. I also thought about redirecting all the output from the Lifecycle start into `/var/log/cassandra/system.log` , so cass-operator's `server-system-logger` would pick it up, but I think it's probably best to keep the process startup logs separate from Cassandra and Management API logs.

I can provide the doc updates, or at least a draft.

**Comment by jsanda** [comment]
After spending more time dealing with some failures and thinking about this I am in favor of consolidating logs into a single file. If Cassandra was running in its own container, `kubectl logs` would return whatever is going to stdout and stderr. I think a single file makes sense. It would be nice if we could delimit the output to improve readability.

**Comment by johnsmartco** [comment]
Gents, @emerkle826 & @jsanda, let me know if you:

* implemented a single log file (if so, where can the user find it, and/or where log entries for C* startup errors would be displayed)
* improved the output with a delimiter
* wrote a draft doc

Many thanks!

**Comment by emerkle826** [comment]
@johnsmartco I haven't implemented any of the suggestions yet for C* logs, nor have I started a doc. I think the work can be handled in the ticket [here](https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/97), but that hasn't bubbled up yet.
